adaptive_mask_inpainting_example.py

```diff
@@ -0,0 +1,180 @@
+        output_file = "Base-PointRend-RCNN-FPN.yaml"
+    )
+    download_file(
+        url = "https://huggingface.co/datasets/jellyheadnadrew/adaptive-mask-inpainting-test-images/resolve/main/Base-RCNN-FPN.yaml?download=true",
+        output_file = "Base-RCNN-FPN.yaml",
+    )
+    
+    """ 
+    Prepare Adaptive Mask Inpainting Pipeline
+    """
+    # device
+    device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+    num_steps = 50
+    
+    # Scheduler
+    scheduler = DDIMScheduler(
+        beta_start=0.00085, 
+        beta_end=0.012, 
+        beta_schedule="scaled_linear", 
+        clip_sample=False, 
+        set_alpha_to_one=False
+    )
+    scheduler.set_timesteps(num_inference_steps=num_steps)
+
+    ## load models as pipelines
+    pipeline = AdaptiveMaskInpaintPipeline.from_pretrained(
+        "Uminosachi/realisticVisionV51_v51VAE-inpainting", 
+        scheduler=scheduler, 
+        torch_dtype=torch.float16, 
+        requires_safety_checker=False
+    ).to(device)
+
+    ## disable safety checker
+    enable_safety_checker = False
+    if not enable_safety_checker:
+        pipeline.safety_checker = None
+
+    # declare segmentation model used for mask adaptation
+    use_visualizer = False
+    assert not use_visualizer, \
+    """
+    If you plan to 'use_visualizer', USE WITH CAUTION. 
+    It creates a directory of images and masks, which is used for merging into a video.
+    The procedure involves deleting the directory of images, which means that 
+    if you set the directory wrong you can have other important files blown away.
+    """
+    
+    adaptive_mask_model = PointRendPredictor(
+        pointrend_thres=0.2, 
+        device="cuda" if torch.cuda.is_available() else "cpu", 
+        use_visualizer=use_visualizer,
+        config_pth="pointrend_rcnn_R_50_FPN_3x_coco.yaml",
+        weights_pth="model_final_edd263.pkl",
+    )
+    pipeline.register_adaptive_mask_model(adaptive_mask_model)
+
+    step_num = int(num_steps * 0.1)
+    final_step_num = num_steps - step_num * 7
+    # adaptive mask settings
+    adaptive_mask_settings = EasyDict(
+        dict(
+            dilate_scheduler=MaskDilateScheduler(
+                max_dilate_num=20,
+                num_inference_steps=num_steps,
+                schedule=[20] * step_num + [10] * step_num + [5] * step_num + [4] * step_num + [3] * step_num + [2] * step_num + [1] * step_num + [0] * final_step_num
+            ),
+            dilate_kernel=np.ones((3, 3), dtype=np.uint8),
+            provoke_scheduler=ProvokeScheduler(
+                num_inference_steps=num_steps,
+                schedule=list(range(2, 10 + 1, 2)) + list(range(12, 40 + 1, 2)) + [45],
+                is_zero_indexing=False,
+            ),
+        )
+    )
+    pipeline.register_adaptive_mask_settings(adaptive_mask_settings)
```

Comment:
same as before

Reply:
Thanks for the thorough feedback! I've applied them, and will PR again.

(Please check minor questions above)